api/app/Http/Transformers/IlluminateModelTransformer.php

```diff
@@ -0,0 +1,164 @@
+                $array = $this->renameArrayKey($array, $key, camel_case($key));
+            }
+        }
+
+        if (($object instanceof BaseModel)) {
+            $this->addSelfKey($object, $array);
+        }
+
+        return $array;
+    }
+
+    /**
+     * Recursive adding of self key
+     * @param BaseModel $model
+     * @param $array
+     */
+    protected function addSelfKey(BaseModel $model, &$array)
+    {
+        if ($route = RouteHelper::getRoute($model)) {
+            $array['_self'] = $route;
+        }
+
+        foreach ($model->getRelations() as $key => $value) {
+            $camelCaseKey = camel_case($key);
+            if ($value instanceof BaseModel && isset($array[$camelCaseKey])) {
+                $this->addSelfKey($value, $array[$camelCaseKey]);
+            } elseif ($this->isIterable($value)) {
+                foreach ($value as $index => $relatedModel) {
+                    if ($relatedModel instanceof BaseModel && isset($array[$camelCaseKey][$index])) {
+                        $this->addSelfKey($relatedModel, $array[$camelCaseKey][$index]);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * @param $var
+     * @return bool
+     */
+    protected function isIterable($var)
+    {
+        return (is_array($var) || $var instanceof Traversable);
+    }
+
+    /**
+     * Rename an array key while preserving array order.
+     *
+     * @param  array   $array
+     * @param  string  $from
+     * @param  string  $to
+     * @return array
+     */
+    protected function renameArrayKey(array $array, $from, $to)
+    {
+        $keys = array_keys($array);
+        $index = array_search($from, $keys);
+
+        if ($index !== false) {
+            $keys[$index] = $to;
+            $array = array_combine($keys, $array);
+        }
+
+        return $array;
+    }
+
+    /**
+     * Recursively rename keys in nested arrays.
+     *
+     * @param  array  $array
+     * @return array
+     */
+    protected function renameKeys(array $array)
+    {
+        $newArray = [];
+        foreach ($array as $key => $value) {
+
+            // Recursively check if the value is an array that needs parsing too
+            $value = (is_array($value)) ? $this->renameKeys($value) : $value;
+
+            // Convert snake_case to camelCase
+            if (is_string($key) && str_contains($key, '_')) {
+                $newArray[camel_case($key)] = $value;
+            } else {
+                $newArray[$key] = $value;
+            }
+        }
+
+        return $newArray;
+    }
+
+    /**
+     * @param $collection
+     * @return mixed
+     */
+    public function transformCollection($collection)
+    {
+        if ($collection instanceof Collection) {
+            $collection = $collection->all();
+        }
+        return $this->getService()->collection(($collection instanceof Collection)?$collection->all():$collection, $this);
```

Comment:
Can you split this large ternary out into an if/else? Avoid ternaries unless it is trivial.